src/cargo/ops/cargo_compile.rs

```diff
@@ -163,6 +167,23 @@ pub fn compile_pkg(package: &Package, options: &CompileOptions)
     let to_build = packages.iter().find(|p| p.package_id() == pkgid).unwrap();
     let targets = try!(generate_targets(to_build, mode, filter, release));
 
+    let target_with_args = match target_rustc_args {
+        &Some(args) => {
+            if targets.len() > 1 {
+                return Err(human("extra arguments to `rustc` can only be \
+                                  invoked for one target"))
+            }
+            let (target, profile) = targets[0];
+            let mut profile = profile.clone();
+            profile.rustc_args = Some(args.to_vec());
+            Some((target, profile))
+        },
+        &None => None,
+    };
+
+    let targets = target_with_args.as_ref().map(|&(t, ref p)| vec!((t, p)))
```

Comment:
Hm, I'm not exactly sure how this would be done since I'm already cloning the `Profile` inside the `match`.
Maybe there's a better way of doing this that does not require the extra `clone()`?